packages/explat-client/src/test/create-explat-client.ts

```diff
@@ -0,0 +1,697 @@
+		Array [
+		  Array [
+		    Object {
+		      "experimentName": "the-invalid-experiment-name",
+		      "message": "Invalid experimentName: the-invalid-experiment-name",
+		    },
+		  ],
+		]
+	` );
+	} );
+} );
+
+describe( 'ExPlatClient.loadExperimentAssignment multiple-use', () => {
+	it( `should respect the ttl`, async () => {
+		const client = createExPlatClient( createMockedConfig() );
+
+		mockFetchExperimentAssignmentToMatchExperimentAssignment( validExperimentAssignment );
+		const firstDate = Date.now();
+		spiedMonotonicNow.mockImplementation( () => firstDate );
+		const experimentAssignmentA = client.loadExperimentAssignment(
+			validExperimentAssignment.experimentName
+		);
+		const experimentAssignmentB = client.loadExperimentAssignment(
+			validExperimentAssignment.experimentName
+		);
+		const experimentAssignmentC = client.loadExperimentAssignment(
+			validExperimentAssignment.experimentName
+		);
+		const experimentAssignmentD = client.loadExperimentAssignment(
+			validExperimentAssignment.experimentName
+		);
+		expect( await experimentAssignmentA ).toEqual( await experimentAssignmentB );
+		expect( await experimentAssignmentA ).toEqual( await experimentAssignmentC );
+		expect( await experimentAssignmentA ).toEqual( await experimentAssignmentD );
+		expect( mockedFetchExperimentAssignment.mock.calls ).toHaveLength( 1 );
+		expect( mockedLogError.mock.calls ).toMatchInlineSnapshot( `Array []` );
+
+		const dateIncreasePastTtl = validExperimentAssignment.ttl * 1000 + 1;
+		const refreshedValidExperimentAssignment = {
+			...validExperimentAssignment,
+			variationName: null,
+			retrievedTimestamp: Date.now() + dateIncreasePastTtl,
+		};
+		mockFetchExperimentAssignmentToMatchExperimentAssignment( refreshedValidExperimentAssignment );
+		spiedMonotonicNow.mockImplementation( () => Date.now() + dateIncreasePastTtl );
+		const experimentAssignmentE = client.loadExperimentAssignment(
+			validExperimentAssignment.experimentName
+		);
+		const experimentAssignmentF = client.loadExperimentAssignment(
+			validExperimentAssignment.experimentName
+		);
+		const experimentAssignmentG = client.loadExperimentAssignment(
+			validExperimentAssignment.experimentName
+		);
+		const experimentAssignmentH = client.loadExperimentAssignment(
+			validExperimentAssignment.experimentName
+		);
+		expect( await experimentAssignmentE ).toEqual( await experimentAssignmentF );
+		expect( await experimentAssignmentE ).toEqual( await experimentAssignmentG );
+		expect( await experimentAssignmentE ).toEqual( await experimentAssignmentH );
+		expect( mockedFetchExperimentAssignment.mock.calls ).toHaveLength( 2 );
+		expect( ( await experimentAssignmentA ).retrievedTimestamp ).toBeLessThan(
+			( await experimentAssignmentE ).retrievedTimestamp
+		);
+		expect( mockedLogError.mock.calls ).toMatchInlineSnapshot( `Array []` );
+	} );
+
+	it( `[developmentMode] should respect the ttl`, async () => {
```

Comment:
This is almost identical to the previous test. Is the behaviour supposed to be the same? It'd be good to reduce duplication here to make the intention clearer.